File: src/main/java/com/owncloud/android/lib/resources/notifications/models/RichObject.java
/*  Nextcloud Android Library is available under MIT license
 *
 *   @author Andy Scherzinger
 *   Copyright (C) 2017 Andy Scherzinger
 *   Copyright (C) 2017 Nextcloud GmbH
 *
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *   of this software and associated documentation files (the "Software"), to deal
 *   in the Software without restriction, including without limitation the rights
 *   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *   copies of the Software, and to permit persons to whom the Software is
 *   furnished to do so, subject to the following conditions:
 *
 *   The above copyright notice and this permission notice shall be included in
 *   all copies or substantial portions of the Software.
 *
 *   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
 *   EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
 *   MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
 *   NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
 *   BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
 *   ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
 *   CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 *   THE SOFTWARE.
 *
 */

package com.owncloud.android.lib.resources.notifications.models;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * Rich object data model providing rich data to be used within rich content, e.g.
 * in {@link Notification}s.
 */
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class RichObject {
    /**
     * A unique identifier for the object type.
     */
    public String type;

    /**
     * A short identifier of the object on the server (int or string).
     */
    public String id;

    /**
     * A name which should be used in the visual representation.
     */
    public String name;
}

File: library/src/main/java/com/owncloud/android/lib/resources/notifications/models/RichObject.kt
/*
 * Nextcloud Android Library
 *
 * SPDX-FileCopyrightText: 2017-2024 Nextcloud GmbH and Nextcloud contributors
 * SPDX-FileCopyrightText: 2017 Andy Scherzinger <info@andy-scherzinger.de>
 * SPDX-License-Identifier: MIT
 */
package com.owncloud.android.lib.resources.notifications.models

/**
 * Rich object data model providing rich data to be used within rich content, e.g.
 * in [Notification]s.
 */
class RichObject() {
    /**
     * A unique identifier for the object type.
     */
    var type: String? = null

    /**
     * A short identifier of the object on the server (int or string).
     */
    @JvmField
    var id: String? = null

    /**
     * A name which should be used in the visual representation.
     */
    var name: String? = null

    constructor(type: String?, id: String?, name: String?) : this() {
        this.type = type
        this.id = id
        this.name = name
    }
}
